src/fprime/common/models/serialize/bool_type.py

```diff
@@ -22,24 +22,25 @@ class BoolType(ValueType):
     TRUE = 0xFF
     FALSE = 0x00
 
-    def validate(self, val):
+    @classmethod
+    def validate(cls, val):
         """Validate the given class"""
         if not isinstance(val, bool):
             raise TypeMismatchException(bool, type(val))
 
     def serialize(self):
         """Serialize a boolean value"""
-        if self.val is None:
+        if self._val is None:
```

Comment:
Bool needs to define a value property and setter.

Reply:
We decided it did not.  The specification is this:
- Primitives set `._val` to a python type. `.val` and `._val` are therefore interchangeable.
- Complex items set `._val` to a python type containing members of `BaseType` and `.val` returns fully-python type.

As long as `BoolType` is self consistent, everything is ok.